pkg/controller/watermarkpodautoscaler/replica_calculator.go

```diff
@@ -17,14 +17,17 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
 	v1coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
 	metricsclient "k8s.io/kubernetes/pkg/controller/podautoscaler/metrics"
 )
 
 // ReplicaCalculatorItf interface for ReplicaCalculator
 type ReplicaCalculatorItf interface {
 	GetExternalMetricReplicas(logger logr.Logger, currentReplicas int32, metric v1alpha1.MetricSpec, wpa *v1alpha1.WatermarkPodAutoscaler) (replicaCount int32, utilization int64, timestamp time.Time, err error)
+	GetResourceReplicas(logger logr.Logger, currentReplicas int32, metric v1alpha1.MetricSpec, wpa *v1alpha1.WatermarkPodAutoscaler) (replicaCount int32, utilization int64, timestamp time.Time, err error)
```

Comment:
I know you are just adding a similar method but it feels like maybe returning a struct for the first three parameters `replicaCount`, `utilization` and `timestamp` could help with readability. It would also help with handling error cases in the implementation.